dev/Interop/CameraCaptureUI/CameraCaptureUI/CameraCaptureUIPhotoCaptureSettings.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#pragma once
+#include "Microsoft.Windows.Media.Capture.CameraCaptureUIPhotoCaptureSettings.g.h"
+#include "PropertyHelper.h"
+
+namespace winrt::Microsoft::Windows::Media::Capture::implementation
+{
+    struct CameraCaptureUIPhotoCaptureSettings : CameraCaptureUIPhotoCaptureSettingsT<CameraCaptureUIPhotoCaptureSettings>
+    {
+        CameraCaptureUIPhotoCaptureSettings() = default;
+
+        void validate();
+
+        hstring GetFileExtension();
+        void Serialize(winrt::Windows::Foundation::Collections::ValueSet const& props);
+
+        basic_property<bool> AllowCropping{ true };
+        basic_property<winrt::Windows::Foundation::Size> CroppedAspectRatio{ 0.0f, 0.0f };
+        basic_property<winrt::Windows::Foundation::Size> CroppedSizeInPixels{ 0.0f, 0.0f };
+        basic_property<winrt::Microsoft::Windows::Media::Capture::CameraCaptureUIPhotoFormat> Format{ winrt::Microsoft::Windows::Media::Capture::CameraCaptureUIPhotoFormat::Jpeg };
+        basic_property<winrt::Microsoft::Windows::Media::Capture::CameraCaptureUIMaxPhotoResolution> MaxResolution{ winrt::Microsoft::Windows::Media::Capture::CameraCaptureUIMaxPhotoResolution::HighestAvailable };
+    };
+}
+
+namespace winrt::Microsoft::Windows::Media::Capture::factory_implementation
+{
+    struct CameraCaptureUIPhotoCaptureSettings : CameraCaptureUIPhotoCaptureSettingsT<CameraCaptureUIPhotoCaptureSettings, implementation::CameraCaptureUIPhotoCaptureSettings>
```

Comment:
Its been long enough that I forgot some details, but is this factory supposed to be non-agile so it has thread affinity?  Or is it fine to be agile and the API is multi-thread safe it just needs to be called from a UI thread?

Reply:
Yes, the factory can be designed to be agile, meaning it can operate across multiple threads, as long as it adheres to the requirement of being called from a UI thread. 

Can you elaborate more on "thread affinity" in this context and your concerns regarding it.

Reply:
Thread affinity means that it can only be used by a specific thread.  It sounds like that is not what you want here, though.  Agile is fine if it can be used from multiple threads (assuming those threads are "UI" threads).